### DISCUSSION TITLE
[Question][Hw5]

### DISCUSSION THREAD

**Discussion by shyngys-aitkazinov** [prosyslab-classroom/cs348-information-security#607] [Q&A]
I am confused what are the phi nodes . Particularly, I am confused with
![image](https://user-images.githubusercontent.com/56180616/120141619-a3451680-c1fe-11eb-9d46-c9f972fec3f4.png)
As far as I understood the phi nodes is like a grid of Es. (We have Eij) But in the implementation of the phi nodes they are just the list of instrs? Can anybody explain why this is so? 

And what is k in this? (Eik)

**Comment by KihongHeo** [comment]
A node of type `Phi of Llvm.llvalue list` consists of a list of phi instructions like
```
x1 = phi(e11, e12) 
x2 = phi(e21, e22)
...
xi = phi(ei1, ei2, ... eik)
```

**Comment by RiceBiscuits** [comment]
this is an example of phi instruction in an IR file. 
%i.0 = phi i32 [ **10**, %entry ], [ **%dec**, %for.inc ], !dbg !16
%sum.0 = phi i32 [ **0**, %entry ], [ **%add**, %for.inc ], !dbg !12

Each phi instruction denotes x, and values (bold style)  in phi instructions are E. 
In this example, x_0 is %i.0, x_1 is %sum.0. 
Also, E_01 is **10**, E_02 is **%dec**.

**Comment by shyngys-aitkazinov** [comment]
ok, thank you very much
